src/librbd/AioImageRequest.cc

```diff
@@ -462,8 +462,15 @@ void AioImageFlush::send_request() {
     m_aio_comp->associate_journal_event(journal_tid);
   }
 
-  C_AioRequest *req_comp = new C_AioRequest(cct, m_aio_comp);
-  m_image_ctx.flush(req_comp);
+  if (!journaling && m_image_ctx.object_cacher != NULL)
```

Comment:
No -- if the cache is disabled, you still might have in-flight writes/discards that you would need to ensure complete safely before finishing the flush.  The ```ImageCtx::flush``` call handles both cases.